lib/cbuffer.c

```diff
@@ -17,7 +17,7 @@
 #include "cbuffer.h"
 
 
-size_t lib_cbufSize(const cbuffer_t *buf)
+__attribute__((section(".noxip"))) size_t lib_cbufSize(const cbuffer_t *buf)
```

Comment:
The `.noxip` section is defined in linker script only on arm. On other archs it's an [orphan section](https://sourceware.org/binutils/docs/ld/Orphan-Sections.html) (and it's up to linker to do sth with it). Even though it's handled correctly (I've checked readelf on sparc) I think it'd be safer to place this section explicitly on other archs in `.text` especially since we rely on `__etext` symbol.